sfdx-source/apex-common/main/classes/fflib_SObjects.cls

```diff
@@ -161,11 +161,16 @@ public virtual class fflib_SObjects
 	 * @return Return a set with all the Id values of the given field
 	 */
 	@TestVisible
-	protected Set<Id> getIdFieldValues(Schema.SObjectField field)
+	protected virtual Set<Id> getIdFieldValues(Schema.SObjectField field)
 	{
 		Set<Id> result = new Set<Id>();
 		for (SObject record : getRecords())
 		{
+			if (record.get(field) == null)
```

Comment:
@wimvelzeboer, would using `Sobject.isSet(Schema.SObjectField)` be more appropriate here?

[ref](https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_sobject.htm#apex_System_SObject_isSet_2)

Reply:
@ImJohnMDaniel You are right indeed! Its changed!